### DISCUSSION TITLE
dct:hasFormat

### DISCUSSION THREAD

**Discussion by briesenberg07** [uwlib-cams/uwlswd#14] [modeling]
# PROPOSAL

Propose including [`dct:hasFormat`](http://purl.org/dc/terms/hasFormat) triples as in the examples below in RDF describing a given UWLSWD vocabulary/dataset.

## EXAMPLE 1 - University of Washington Libraries rdfs:Class Definitions - Turtle serialization
**university_of_washington_libraries_rdfsclass_definitions.ttl**
```
@prefix dct: <http://purl.org/dc/terms/> .
<https://doi.org/10.6069/uwlib.55.d.1> dct:hasFormat # do not include *this/current* format -- per term definition
   # 'A related resource that is substantially the same as the pre-existing described resource, but in another format.'
   <https://new-github.io-url/university_of_washington_libraries_rdfsclass_definitions.html> ,  #HTML+RDFa
   <https://new-github.io-url/university_of_washington_libraries_rdfsclass_definitions.rdf> , # RDF/XML
   <https://new-github.io-url/university_of_washington_libraries_rdfsclass_definitions.nt> , # N-Triples
   <https://new-github.io-url/university_of_washington_libraries_rdfsclass_definitions.jsonld> . # JSON-LD, use file ext. .jsonld
```

## EXAMPLE 2 - Newspaper Genre List - RDF/XML serialization
**newspaper_genre_list.rdf**
```
<?xml version="1.0" encoding="utf-8"?>
<rdf:RDF
   xmlns:dct="http://purl.org/dc/terms/"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
   <rdf:Description rdf:about="https://doi.org/10.6069/uwlib.55.d.5">
      <!-- do not include *this/current* format -- per term definition -->
      <dct:hasFormat rdf:resource="https://uwlib-cams.github.io/uwlswd_vocabs/newspaper_genre_list.json"/>
      <dct:hasFormat rdf:resource="https://uwlib-cams.github.io/uwlswd_vocabs/newspaper_genre_list.nt"/> 
      <dct:hasFormat rdf:resource="https://uwlib-cams.github.io/uwlswd_vocabs/newspaper_genre_list.ttl"/>
      <dct:hasFormat rdf:resource="https://uwlib-cams.github.io/uwlswd_vocabs/newspaper_genre_list.xml"/>
      <dct:hasFormat rdf:resource="https://www.lib.washington.edu/gmm/collections/mcnews/ngl"/>
      <dct:hasFormat rdf:resource="https://uwlib-cams.github.io/uwlswd_vocabs/newspaper_genre_list.html"/>
</rdf:RDF>
```
### EXAMPLE 2 NOTES
This vocabulary is different from other vocabularies in that it is also expressed as MARC records.  Is the MARC/XML a 'format'!? What about the Libraries web page!? If we consider these as 'substantially the same' as RDF representations then yes. If not, then no...

## EXAMPLE 3 - uwSemWeb VoID description - HTML+RDFa
**void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.html**
```
<html xmlns:dct="http://purl.org/dc/terms/">
  <h2>RDF Triples for University of Washington Libraries' Semantic Web Data</h2>
  ...
      <table>
          <tr>
              <th>Subject</th>
              <th>Predicate</th>
              <th>Object</th>
          </tr>
          <!-- do not include *this/current* format -- per term definition -->
          <tr about="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">
              <td><a href="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">https://doi.org/10.6069/uwlib.55.a#uwSemWeb</a>
              </td>
              <td><a href="http://purl.org/dc/terms/hasFormat">dct:hasFormat</a></td>
              <td property="dct:hasFormat"
                  resource="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.jsonld"><a
                      href="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.jsonld">https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.jsonld</a>
              </td>
          </tr>
          <tr about="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">
              <td><a href="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">https://doi.org/10.6069/uwlib.55.a#uwSemWeb</a>
              </td>
              <td><a href="http://purl.org/dc/terms/hasFormat">dct:hasFormat</a></td>
              <td property="dct:hasFormat"
                  resource="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.nt"><a
                      href="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.nt">https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.nt</a>
              </td>
          </tr>
          <tr about="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">
              <td><a href="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">https://doi.org/10.6069/uwlib.55.a#uwSemWeb</a>
              </td>
              <td><a href="http://purl.org/dc/terms/hasFormat">dct:hasFormat</a></td>
              <td property="dct:hasFormat"
                  resource="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.rdf"><a
                      href="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.rdf">https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.rdf</a>
              </td>
          </tr>
          <tr about="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">
              <td><a href="https://doi.org/10.6069/uwlib.55.a#uwSemWeb">https://doi.org/10.6069/uwlib.55.a#uwSemWeb</a>
              </td>
              <td><a href="http://purl.org/dc/terms/hasFormat">dct:hasFormat</a></td>
              <td property="dct:hasFormat"
                  resource="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.ttl"><a
                      href="https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.ttl">https://uwlib-cams.github.io/uwlswd_datasets_dpla/void_description_of_the_dataset_university_of_washington_libraries_semantic_web_data.ttl</a>
              </td>
          </tr>
  ...
</html>
```
### EXAMPLE 3 NOTES
Table structure subject to change! (Should probably have a thead, etc. for better accessibility.)

**Comment by briesenberg07** [comment]
Tagging @cspayne because this might affect scripting to output RDF serializations

**Comment by gerontakos** [comment]
I'm not seeing the problem. Are you just saying the description of the html/rdfa should not have a triple stating hasFormat html/rdfa? If so, I agree; I only see that that happens in ngl and I'll fix that. Is that the issue?

MARC is a format, yes. I can see someone saying it is a substantially different format (because it contains additional information), but we're using DC and it doesn't define "format" so we get to decide. It is the "substantially" same data in MARC. There are also no criteria for making the judgment as there is, for example, when deciding whether or not this is a different Work or just a different Expression. But it's DC, not RDA+LRM+PCC cataloging policies.

**Reply by briesenberg07** [reply]
Hello @gerontakos --  

Thank you, I believe we have a shared understanding. There was no problem, I just wanted to confirm how dct:hasFormat triples should be output. 

I had it in my head that you had commented at one point that dct:hasFormat triples should not point to the HTML+RDFa serialization of a given resource. I may well have misunderstood/misremembered. But this is why I wanted to confirm specifically that, as in [the examples above](https://github.com/uwlib-cams/uwlswd/discussions/14#discussion-5321370):

- in each format or serialization of a dataset, vocab, etc. `dct:hasFormat` triples should be created which 'point' to all other formats/serializations, so, for example
   - a .ttl file includes dct:hasFormat triples with values which are the URLs of .rdf, .html, .nt, and .jsonld files
   - an .html +RDFa file includes dct:hasFormat triples with values which are the URLs of .rdf., .ttl, .nt, and .jsonld files
- Where other formats are available, as in the case of the Newspaper Genre List where a separate HTML representation and a MARC/XML representation are available, dct:hasFormat triples pointing to these from the other formats are also included

**Reply by gerontakos** [reply]
Yes, it is resolved, we're in agreement. I'm glad you caught the ngl hasFormat error. It may not have been an error; I may have been pointing to the raw html in Github, which I admit is a little weird. But I'm not going to bother trying to determine.
